torch/csrc/Module.cpp

```diff
@@ -636,6 +668,22 @@ static PyObject* THPModule_fromDLPack(PyObject* _unused, PyObject* data) {
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject* THPModule_torchDeviceToDLDevice(
+    PyObject* _unused,
+    PyObject* data) {
+  HANDLE_TH_ERRORS
+  TORCH_CHECK(
+      THPDevice_Check(data),
+      "torchDeviceToDLDevice: expected torch.device argument.");
+  auto device = reinterpret_cast<THPDevice*>(data)->device;
+  auto dl_device = at::torchDeviceToDLDevice(device);
+  auto tuple = PyTuple_New(2);
```

Comment:
Add error checking in case this failed

Reply:
Not sure I get what you mean. If `at::torchDeviceToDLDevice` errors, the error will bubble up to Python, won't it?

Reply:
CPython is C API, not C++. There is no error throwing here, you need to check the tuple is not null and if so, make this a c++ error by `throw python_error();` 

```cpp
if (!tuple) {
  throw python_error();
}
```